src/main/kotlin/validation/Validator.kt

```diff
@@ -20,6 +21,12 @@ val EMAIL_REGEX =
     Regex(
         "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)])",
     )
+val SPECIAL_CHAR_REGEX = Regex("[" + Regex.escape("#$%&@^`~.,:;\"'/|_-<>*+!?={[()]}ß") + "]")
+const val PASSWORD_MIN_AMOUNT_LOWERCASE_LETTERS = 2
+const val PASSWORD_MIN_AMOUNT_UPPERCASE_LETTERS = 2
+const val PASSWORD_MIN_AMOUNT_DIGITS = 2
+const val PASSWORD_MIN_AMOUNT_SPECIAL_CHARS = 2
```

Comment:
"amount" is only used for things that cannot be separated into units, such as liquid.
```suggestion
const val PASSWORD_MIN_NUMBER_LOWERCASE_LETTERS = 2
const val PASSWORD_MIN_NUMBER_UPPERCASE_LETTERS = 2
const val PASSWORD_MIN_NUMBER_DIGITS = 2
const val PASSWORD_MIN_NUMBER_SPECIAL_CHARS = 2
```